lib/omniauth/strategies/square.rb

```diff
@@ -30,13 +30,24 @@ def raw_info
         @raw_info ||= access_token.get('/v1/me').parsed
       end
 
+      alias :old_request_phase :request_phase
+
+      def request_phase
+        if request.params['plan_id']
+          options[:authorize_params][:plan_id] = request.params['plan_id']
+        end
+
+        old_request_phase
```

Comment:
@vdolgishev I'm working on some updates to the gem overall, so this should be merged in very soon.